### PR TITLE
PR: QWebChannel implementation and tests

### LINKED ISSUE #102: Adapt close functionallity on Qt WebEngine
PR #98 can't be used on WebEngine due to an unspecified JS ReferenceError (Maybe JS are not loaded completely) which causes to open and close terminal tabs before they are loaded. I think this must be addressed by means of a [QWebChannel](http://doc.qt.io/qt-5/qwebchannel.html)

### PR DESCRIPTION
**Pull request by andfoy**
Fixes #102

### CODE REVIEW

File: spyder_terminal/widgets/terminalgui.py [hunk 1]
```diff
@@ -21,6 +22,29 @@
 from spyder.utils.qthelpers import create_action, add_actions
 
 from qtpy.QtWebEngineWidgets import WEBENGINE
+if WEBENGINE:
+    from PyQt5.QtWebChannel import QWebChannel
```
Comment (ccordoba12) [inline spyder_terminal/widgets/terminalgui.py]: Could we add support for `QtWebChannel` in qtpy? I think it's better because PySide2 also supports Qt5.

Comment (andfoy) [inline spyder_terminal/widgets/terminalgui.py]: I was thinking exactly the same! I can send a PR

File: circle.yml [hunk 2]
```diff
@@ -28,7 +28,7 @@ dependencies:
       export PATH="$HOME/miniconda/bin:$PATH" &&
       source activate test &&
       conda install -q ciocheck -c spyder-ide --no-update-deps &&
-      if [ "$CIRCLE_NODE_INDEX" = "2" ]; then pip install -q spyder pyqt5 tornado pexpect; else conda install -q -y spyder && conda install tornado pexpect; fi &&
+      if [ "$CIRCLE_NODE_INDEX" = "2" ]; then pip install -q spyder pyqt5==5.8.0 tornado pexpect; else conda install -q -y spyder && conda install tornado pexpect; fi &&
```
Comment (ccordoba12) [inline circle.yml]: Why do we need to pin `pyqt5` to `5.8.0`

Comment (andfoy) [inline circle.yml]: I tried with 5.9.0, but it segfaults

File: spyder_terminal/widgets/terminalgui.py (hunk 1 above)
Comment (ccordoba12) [inline spyder_terminal/widgets/terminalgui.py]: Yes, please do it.

File: circle.yml (hunk 2 above)
Comment (ccordoba12) [inline circle.yml]: Ok

**Comment by stonebig** [comment]
I'm in PyQt-5.9.0 / Windows , so I should not upgrade for now ?

**Comment by ccordoba12** [comment]
The problem was only related to our tests.

**Pull request merged**